site/app/controllers/admin/UsersController.php

```diff
@@ -411,7 +411,20 @@ public function updateRegistrationSections() {
             if (User::validateUserData('registration_section', $_POST['delete_reg_section'])) {
                 // DELETE trigger function in master DB will catch integrity violation exceptions (such as FK violations when users/graders are still enrolled in section).
                 // $num_del_sections indicates how many DELETEs were performed.  0 DELETEs means either the section didn't exist or there are users still enrolled.
-                $num_del_sections = $this->core->getQueries()->deleteRegistrationSection($_POST['delete_reg_section']);
+                //why doesn't $this->core->getConfig()->getCoursePath(); work?
```

Comment:
I don't think it makes sense to not delete a registration section if its used in course materials. A better approach would be to assume a registration section could disappear at any time and if its not valid let course materials ignore it and move on. If there are no valid sections just release to everyone in that case.

Reply:
You don't think it would be bad to have references to sections that no longer exist in the json file? It seems like that could cause problems later on if we ever want to build off of this...

Reply:
It would be better if course materials assumes a section could be deleted and handle it accordingly. If an instructor wanted to delet a section it would be more annoying than anything to go into course materials and delete all references.
Instead course materials should be fault tolerant and handle the case if a section does not exist gracefully.

Reply:
Resolving based on meeting - didn't know this was the expected behavior